lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -1619,7 +1650,7 @@ selectRndStateAddresses
 selectRndStateAddresses wid sl = do
     addrs <- fmap entityVal <$> selectList
         [ RndStateAddressWalletId ==. wid
-        , RndStateAddressSlot ==. sl
+        , RndStateAddressSlot <=. sl
```

Comment:
could you explain why this change ?

Reply:
We used to store addresses as "snapshots" such that at every slot we would have the entire address space. 
This PR changes this to store addresses as _diff_, such that each slot only contains newly discovered addresses. This means that in order to find **all** known addresses at a particular slot sl, we need not only addresses discovered at sl, but all addresses we ever discovered.